src/main/java/core/basesyntax/service/impl/DataValidatorImpl.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import core.basesyntax.exception.InvalidFruitTypeException;
+import core.basesyntax.exception.InvalidOperationException;
+import core.basesyntax.exception.InvalidQuantityException;
+import core.basesyntax.service.DataValidator;
+
+public class DataValidatorImpl implements DataValidator {
+    private static final String BALANCE = FruitDataParser.Operation.BALANCE.getCode();
+    private static final String SUPPLY = FruitDataParser.Operation.SUPPLY.getCode();
+    private static final String RETURN = FruitDataParser.Operation.RETURN.getCode();
+    private static final String PURCHASE = FruitDataParser.Operation.PURCHASE.getCode();
+    private static final String SPECIAL_CHARACTERS = ".*[$&+,/:;=?@#|'<>.-^*()%!].*";
+
+    @Override
+    public void validate(FruitTransactionDto fruitTransactionDto) {
+        validateFruitType(fruitTransactionDto);
+        validateOperation(fruitTransactionDto);
+        validateQuantity(fruitTransactionDto);
+
+    }
+
+    @Override
+    public void validateOperation(FruitTransactionDto fruitTransactionDto) {
+        String operation = fruitTransactionDto.operation();
+
+        if ((operation.equalsIgnoreCase(BALANCE)
+                || operation.equalsIgnoreCase(SUPPLY)
+                || operation.equalsIgnoreCase(RETURN)
+                || operation.equalsIgnoreCase(PURCHASE))) {
+            return;
+        }
+        throw new InvalidOperationException("Invalid operation: " + operation);
+    }

```

Comment:
```suggestion
```
let's better remove this operation validation and throw InvalidOperationException , if there is no operation in `Map<Operation, OperationHandler> strategyMap`